blog/_posts/2017-09-04-gsoc-NeuralNetDiffEq.md

```diff
@@ -0,0 +1,172 @@
+
+## Examples
+Below you can find a few examples on how to use the package I've been working on.
+Following are the initial imports required for the package to work.
+```julia
+using NeuralNetDiffEq
+using Plots; plotly()
+using DiffEqBase, ParameterizedFunctions
+using DiffEqProblemLibrary, DiffEqDevTools
+using Knet
+```
+### ODE Examples
+#### Example 1
+
+
+  ```julia
+  linear = (t,u) -> (1.01*u)
+  (f::typeof(linear))(::Type{Val{:analytic}},t,u0) = u0*exp(1.01*t)
+  prob = ODEProblem(linear,1/2,(0.0,1.0))
+  sol = solve(prob,nnode(10),dt=1/10,iterations=10)
+  plot(sol,plot_analytic=true)
+```
+![Plot_ode1](/images/blog/2017-09-04-gsoc-NeuralNetDiffEq/plot_ode1.png "Plot 1")
+```julia
+  sol(0.232)
+  ```
+
+
+
+
+      1-element Array{Any,1}:
+       0.625818
+#### Example 2
+
+```julia
+f = (t,u) -> (t^3 + 2*t + (t^2)*((1+3*(t^2))/(1+t+(t^3))) - u*(t + ((1+3*(t^2))/(1+t+t^3))))
+(::typeof(f))(::Type{Val{:analytic}},t,u0) =  u0*exp(-(t^2)/2)/(1+t+t^3) + t^2
+prob2 = ODEProblem(f,1.0,(0.0,1.0))
+sol2 = solve(prob2,nnode(10),dt=0.1,iterations=200)
+ plot(sol,plot_analytic=true)
+```
+![Plot_ode2](/images/blog/2017-09-04-gsoc-NeuralNetDiffEq/plot_ode2.png "Plot 2")
+```julia
+  sol(0.47)
+  ```
+
+
+
+
+      1-element Array{Any,1}:
+       0.803109
+
+#### Example 3
+```julia
+f2 = (t,u) -> (-u/5 + exp(-t/5).*cos(t))
+(::typeof(f2))(::Type{Val{:analytic}},t,u0) =  exp(-t/5)*(u0 + sin(t))
+prob3 = ODEProblem(f2,Float32(0.0),(Float32(0.0),Float32(2.0)))
+sol3 = solve(prob3,nnode(10),dt=0.2,iterations=1000)
+ plot(sol,plot_analytic=true)
+```
+![Plot_ode3](/images/blog/2017-09-04-gsoc-NeuralNetDiffEq/plot_ode3.png "Plot 3")
+```julia
+  sol3([0.721])
+  ```
+
+
+
+
+      1-element Array{Any,1}:
+        Any[0.574705]
+
+
+### SODE Examples
+#### Example 1  ODE 2D Linear
+
+
+```julia
+f_2dlinear = (t,u) -> begin
+    du = Array{Any}(length(u))
+    for i in 1:length(u)
+    du[i] = 1.01*u[i]
+  end
+    return du
+end
+(f::typeof(f_2dlinear))(::Type{Val{:analytic}},t,u0) = u0*exp.(1.01*t)
+prob_ode_2Dlinear = ODEProblem(f_2dlinear,rand(4,1),(0.0,1.0))
+sol1 = solve(prob_ode_2Dlinear,nnode([10,50]),dt=0.1,iterations=100)
+```
+
+    (:iteration,100,:loss,0.004670103680503722)
+     16.494870 seconds (90.08 M allocations: 6.204 GB, 5.82% gc time)
+
+
+
+```julia
+plot(sol1,plot_analytic=true)
+```
+![Plot_sode1](/images/blog/2017-09-04-gsoc-NeuralNetDiffEq/plot_ode1.png "Plot 1")
+#### Example 2 Lotka Volterra
+
+
+```julia
+function lotka_volterra(t,u)
+  du1 = 1.5 .* u[1] - 1.0 .* u[1].*u[2]
+  du2 = -3 .* u[2] + u[1].*u[2]
+  [du1,du2]
+end
+```
+
+
+
+
+    lotka_volterra (generic function with 1 method)
+
+
+
+
+```julia
+prob_ode_lotkavoltera = ODEProblem(lotka_volterra,Float32[1.0,1.0],(Float32(0.0),Float32(1.0)))
```

Comment:
You should try to solve this on `(0.0,10.0)`. This is a classic oscillating equation and it looks like it's incorrect because the solution isn't oscillating, when in reality its timespan is just too short.

Reply:
(0.0,5.0) is probably enough if it needs to be faster.

Reply:
Do not have an optimize NN for that but will update later when I have.
The graph won't be proper now